Sources/PrinceOfVersions/Common/ConfigurationData.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ConfigurationData.swift
+//  PrinceOfVersions
+//
+//  Created by Ivana Mršić on 16/04/2020.
+//  Copyright © 2020 Infinum Ltd. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - ConfigurationData -
+
+struct ConfigurationData: Decodable {
+    let requiredVersion: Version?
+    let lastVersionAvailable: Version?
+    let notifyLastVersionFrequency: NotificationType?
+    let requirements: Requirements?
+    let meta: [String: AnyDecodable]?
+}
+
+// MARK: - Requirements -
+
+struct Requirements: Decodable {
+
+    // MARK: - Internal properties
+
+    let requiredOSVersion: Version?
+    var userDefinedRequirements: [String: Any]
+
+    var allRequirements: [String: Any]? {
+        guard let requiredOSVersion = requiredOSVersion else { return nil }
+        var requirements: [String : Any] = [:]
+        requirements.updateValue(requiredOSVersion, forKey: CodingKeys.requiredOSVersion.rawValue)
+        return requirements.merging(userDefinedRequirements, uniquingKeysWith: { (_, newValue) in newValue })
```

Comment:
Since `requiredOSVersion` parameter is mandatory, I would rather leave `guard-let`check as the first line in this code chunk for that than switch it up with `if-let`. Other suggestion in this comment I have applied.

Reply:
If `requiredOSVersion` is mandatory, than we should remove the optionality from it's declaration 😊 
`let requiredOSVersion: Version?` -> `let requiredOSVersion: Version`